Consolidate package configuration in pyproject.toml (#152)
RMeli/spyrmsd: CHANGELOG.md

### Added

* Python `3.13` and `3.14` in CI [PR #147 | @RMeli]

> * Optional dependencies for testing (`test`), development (`dev`), and parallelization (`parallel`) in `pyproject.toml` [PR #XXX | @RMeli]

Reviewer: ```suggestion
* Optional dependencies for testing (`test`), development (`dev`), and parallelization (`parallel`) in `pyproject.toml` [PR #152 | @copilot]
```

### Changed

* Migrated all package configuration from `setup.cfg` and `setup.py` to `pyproject.toml` [PR #XXX | @RMeli]

> * Workflows to use `pip install -e .` instead of `python setup.py develop` [PR #XXX | @RMeli]

Reviewer: ```suggestion
* Migrated all package configuration from `setup.cfg` and `setup.py` to `pyproject.toml` [PR #152 | @copilot]
```

### Removed

* support for Open Babel [PR #151 | @RMeli]
* Python `3.9` and `3.10` from CI [PR #147 | @RMeli]
* `setup.cfg` file in favour of `pyproject.toml` [PR #XXX | @RMeli]
* `setup.py` file in favour of `pyproject.toml` [PR #XXX | @RMeli]

> * `.coveragerc` file in favour of `pyproject.toml` [PR #XXX | @RMeli]

Reviewer: ```suggestion
* `setup.cfg` and `setup.py` in favour of `pyproject.toml` [PR #152 | @copilot]
```
